driver-core/src/main/java/com/datastax/driver/core/QueryOptions.java

```diff
@@ -169,4 +170,32 @@ public QueryOptions setDefaultIdempotence(boolean defaultIdempotence) {
     public boolean getDefaultIdempotence() {
         return defaultIdempotence;
     }
+
+    /**
+     * Set whether {@link SessionManager#prepare} should prepare statements once or on all the nodes in the cluster.
+     *
+     * Setting this option to true is useful if you have many clients and many nodes and
+     * each client prepares the same statement at startup, in this case re-preparing each
+     * time on all nodes can be redundant.
+     *
+     * @param prepareStatementsOnce the new value to set to indicate whether to prepare
+     *                              statements once or on all nodes.
+     * @return this {@code QueryOptions} instance.
+     */
+    public QueryOptions setPrepareStatementsOnce(boolean prepareStatementsOnce) {
+        this.prepareStatementsOnce = prepareStatementsOnce;
+        return this;
+    }
+
+    /**
+     * The value to indicate if statements are prepared once or on all nodes.
+     * <p/>
+     * The default behaviour is to prepare on all nodes (value is false).
+     *
+     * @return the value which indicates whether statements are prepared once
+     *         or on all nodes.
+     */
+    public boolean getPrepareStatementsOnce() {
```

Comment:
Nitpicking: the getter of a primitive boolean property should start with "is": `isPrepareStatementsOnce()` sounds better here (with `defaultIdempotence`, I'd say we have an exception to this rule in that the property itself is not true or false, it simply tells if the default value for idempotence should be true or false).